### PR TITLE
Fix `ContactPatientBottomSheet` not going back to call patient view on back click in call later mode

### PR DESCRIPTION
**Pull request by msasikanth**
**Testing steps**:
- Go to overdue list
- Open contact patient bottom sheet
- Select "Remind to call later..."
- Click Android back button
- It should go back to call patient view

### CODE REVIEW

**Comment by msasikanth** [comment]
Unfortunately [`OnBackPressedCallback`](https://developer.android.com/reference/androidx/activity/OnBackPressedCallback), doesn't work in this scenario (for dialogs). So we would have to use `setOnKeyListener`.

File: app/src/main/java/org/simple/clinic/util/AndroidExtensions.kt [hunk 1]
```diff
@@ -64,3 +64,14 @@ private fun Configuration.isLocaleAlreadyOverriden(): Boolean {
     else -> false
   }
 }
+
+inline fun Dialog.onBackPressed(crossinline backPressed: () -> Unit) {
+  setOnKeyListener { _, keyCode, event ->
+    if (event.action == KeyEvent.ACTION_UP && keyCode == KeyEvent.KEYCODE_BACK) {
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/util/AndroidExtensions.kt:70]: I feel like this condition `event.action == KeyEvent.ACTION_UP && keyCode == KeyEvent.KEYCODE_BACK` is being used in multiple places. Should we move this to an extension property, say, `KeyEvent.isBackPressEvent` or something?

**Review by vinaysshenoy — CHANGES_REQUESTED** [review]
Overall PR looks good, left one comment.

File: app/src/main/java/org/simple/clinic/util/AndroidExtensions.kt (hunk 1 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/util/AndroidExtensions.kt:70]: Actually, this is the only place we are using it now, I have removed the previous `onOverrideKey` extension. So for sheets that want to handle the back clicks manually, they need to use this extension.

**Review by vinaysshenoy — APPROVED** [review]

**Pull request merged**